de.metas.adempiere.adempiere/base/src/test/java/de/metas/location/impl/AddressBuilderTest.java

```diff
@@ -14,14 +14,23 @@
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
 
+import de.metas.ShutdownListener;
+import de.metas.StartupListener;
 import de.metas.adempiere.model.I_AD_User;
 import de.metas.bpartner.service.IBPartnerBL;
 import de.metas.bpartner.service.impl.BPartnerBL;
+import de.metas.greeting.GreetingRepository;
 import de.metas.interfaces.I_C_BPartner;
 import de.metas.organization.OrgId;
 import de.metas.user.UserRepository;
+import lombok.Builder;
 
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = { StartupListener.class, ShutdownListener.class, GreetingRepository.class })
```

Comment:
Don't use SpringRunner. Just use org.compiere.SpringContextHolder.registerJUnitBean(T) to register ur bean (in ur case, register the GreetingRepository).

More, when touching a junit4 test, pls convert it to junit5.

Reply:
@teosarca then, can we change the message from here?
`org.compiere.SpringContextHolder.throwExceptionIfNull(ApplicationContext)`
It is not obsolete? I did changes reading that message.

Reply:
yes, he have to change the message. it's totally obsolete.